tracing/src/main/java/com/palantir/remoting2/tracing/AsyncSlf4jSpanObserver.java

```diff
@@ -95,6 +98,13 @@ static ZipkinCompatSpan fromSpan(Span span, ZipkinCompatEndpoint endpoint) {
             return annotations;
         }
 
+        private static Iterable<? extends ZipkinCompatBinaryAnnotation> spanMetadataToZipkinBinaryAnnotations(
+                Span span, ZipkinCompatEndpoint endpoint) {
+            return span.getMetadata().entrySet().stream()
+                    .map(entry -> ZipkinCompatBinaryAnnotation.of(entry.getKey(), entry.getValue(), endpoint))
+                    .collect(Collectors.toList());
```

Comment:
changed to loop. I didn't instantiate List with fixed size though since it's unknown.